scripts/release-post.js

```diff
@@ -211,18 +211,29 @@ function writeToFile(results) {
   );
 
   return new Promise((resolve, reject) => {
-    if (fs.existsSync(filepath) && process.argv[3] !== '--force') {
```

Comment:
This script is used (maybe not anymore?) by the releasers to create their blog posts. I don't think these changes are required. Also not clear why the execute bit is being removed

Reply:
> This script is used (maybe not anymore?) by the releasers to create their blog posts.

This script is still used by @nodejs/releasers. It's referenced in https://github.com/nodejs/node/blob/main/doc/contributing/releases.md#17-create-a-blog-post.

Reply:
> not clear why the execute bit is being removed

I'm not sure whether you mean if I've removed:

```js
if (fs.existsSync(filepath) && process.argv[3] !== '--force') {
```

If so, in fact according to codeQL's suggestion, I used "openSync" to check whether the file is "readable" or not (whether the file exists or not), and if yes without "--force", the file cannot be overridden.

PS: To be honest, it's not easy to process with 'fs.openSync' totally, but now everthing goes fine after my tests locally.

Reply:
> not clear why the execute bit is being removed

I'm not sure whether you mean if I've removed:

```js
if (fs.existsSync(filepath) && process.argv[3] !== '--force') {
```

If so, in fact according to codeQL's suggestion, I used "openSync" to check whether the file is "readable" or not (whether the file exists or not), and if yes without "--force", the file cannot be overridden.

PS: To be honest, it's not easy to process with 'fs.openSync' totally, but now everthing goes fine after my tests locally.